modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -1518,6 +1518,31 @@ TEST_P(Test_ONNX_layers, Einsum_const_inputs) {
     testONNXModels("einsum_const_inputs", npy, 0, 0, false, false, 1);
 }
 
+TEST_P(Test_ONNX_layers, Cuda_concat)
+{
+    String basename = "matmul_concat";
+    Net net = readNetFromONNX(_tf("models/" + basename + ".onnx"));
+    Net net_cuda = readNetFromONNX(_tf("models/" + basename + ".onnx"));
+
+    ASSERT_FALSE(net.empty());
+    ASSERT_FALSE(net_cuda.empty());
+
+    net.setPreferableBackend(backend);
+    net.setPreferableTarget(target);
```

Comment:
Here `net` also runs different backends. I think we want it run only in the CPU backend and compare results against the ones in the CUDA backend, right? If so, I propose to move it to `test_backends.cpp`.

Reply:
Agree. And no need for reference input/output data. Let's run on CPU and compare with backend.

Reply:
This is already handled via https://github.com/opencv/opencv/pull/24834/files#diff-94c1c80543469b3a5e5c3276dcb55c5d47acf120f977d31f3d1bfaadc4011850R1531.